### ISSUE TITLE
WebhookPayloadWorkflowRun is incomplete

### ISSUE THREAD

**Issue by matfax** [octokit/webhooks.js#303]
Currently, `WebhookPayloadWorkflowRun` contains the `action`, `organization`, `repository`, and `sender`.
What seems to be missing, however, are the `workflow` and the `workflow_run` themselves.
Is there any particular reason for this?

<details>
  <summary>Here is an example payload of mine for these two missing fields:</summary>

```json
  "workflow": {
    "badge_url": "https://github.com/gofunky/interactive-event-dispatcher/workflows/test/badge.svg",
    "created_at": "2020-10-02T12:42:30.000Z",
    "html_url": "https://github.com/gofunky/interactive-event-dispatcher/blob/master/.github/workflows/test.yml",
    "id": 2823525,
    "name": "test",
    "node_id": "MDg6V29ya2Zsb3cyODIzNTI1",
    "path": ".github/workflows/test.yml",
    "state": "active",
    "updated_at": "2020-10-03T19:24:48.000Z",
    "url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/actions/workflows/2823525"
  },
  "workflow_run": {
    "artifacts_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/actions/runs/289782451/artifacts",
    "cancel_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/actions/runs/289782451/cancel",
    "check_suite_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/check-suites/1291536064",
    "conclusion": "success",
    "created_at": "2020-10-05T16:33:24Z",
    "event": "repository_dispatch",
    "head_branch": "master",
    "head_commit": {
      "author": {
        "email": "matthias.fax@gmail.com",
        "name": "Matthias Fax"
      },
      "committer": {
        "email": "noreply@github.com",
        "name": "GitHub"
      },
      "id": "3484a3fb816e0859fd6e1cea078d76385ff50625",
      "message": "build(test): check workflow run event payload",
      "timestamp": "2020-10-05T16:32:07Z",
      "tree_id": "c30f6c3377009c4c91c85d1f15deb6f2b7c2eb37"
    },
    "head_repository": {
      "archive_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/{archive_format}{/ref}",
      "assignees_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/assignees{/user}",
      "blobs_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/git/blobs{/sha}",
      "branches_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/branches{/branch}",
      "collaborators_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/collaborators{/collaborator}",
      "comments_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/comments{/number}",
      "commits_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/commits{/sha}",
      "compare_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/compare/{base}...{head}",
      "contents_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/contents/{+path}",
      "contributors_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/contributors",
      "deployments_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/deployments",
      "description": "an effort to correct various shortcomings of the default GitHub action and event system",
      "downloads_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/downloads",
      "events_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/events",
      "fork": false,
      "forks_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/forks",
      "full_name": "gofunky/interactive-event-dispatcher",
      "git_commits_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/git/commits{/sha}",
      "git_refs_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/git/refs{/sha}",
      "git_tags_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/git/tags{/sha}",
      "hooks_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/hooks",
      "html_url": "https://github.com/gofunky/interactive-event-dispatcher",
      "id": 300029405,
      "issue_comment_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/issues/comments{/number}",
      "issue_events_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/issues/events{/number}",
      "issues_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/issues{/number}",
      "keys_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/keys{/key_id}",
      "labels_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/labels{/name}",
      "languages_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/languages",
      "merges_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/merges",
      "milestones_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/milestones{/number}",
      "name": "interactive-event-dispatcher",
      "node_id": "MDEwOlJlcG9zaXRvcnkzMDAwMjk0MDU=",
      "notifications_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/notifications{?since,all,participating}",
      "owner": {
        "avatar_url": "https://avatars3.githubusercontent.com/u/41548062?v=4",
        "events_url": "https://api.github.com/users/gofunky/events{/privacy}",
        "followers_url": "https://api.github.com/users/gofunky/followers",
        "following_url": "https://api.github.com/users/gofunky/following{/other_user}",
        "gists_url": "https://api.github.com/users/gofunky/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/gofunky",
        "id": 41548062,
        "login": "gofunky",
        "node_id": "MDEyOk9yZ2FuaXphdGlvbjQxNTQ4MDYy",
        "organizations_url": "https://api.github.com/users/gofunky/orgs",
        "received_events_url": "https://api.github.com/users/gofunky/received_events",
        "repos_url": "https://api.github.com/users/gofunky/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/gofunky/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/gofunky/subscriptions",
        "type": "Organization",
        "url": "https://api.github.com/users/gofunky"
      },
      "private": false,
      "pulls_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/pulls{/number}",
      "releases_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/releases{/id}",
      "stargazers_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/stargazers",
      "statuses_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/statuses/{sha}",
      "subscribers_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/subscribers",
      "subscription_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/subscription",
      "tags_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/tags",
      "teams_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/teams",
      "trees_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/git/trees{/sha}",
      "url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher"
    },
    "head_sha": "3484a3fb816e0859fd6e1cea078d76385ff50625",
    "html_url": "https://github.com/gofunky/interactive-event-dispatcher/actions/runs/289782451",
    "id": 289782451,
    "jobs_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/actions/runs/289782451/jobs",
    "logs_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/actions/runs/289782451/logs",
    "node_id": "MDExOldvcmtmbG93UnVuMjg5NzgyNDUx",
    "pull_requests": [],
    "repository": {
      "archive_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/{archive_format}{/ref}",
      "assignees_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/assignees{/user}",
      "blobs_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/git/blobs{/sha}",
      "branches_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/branches{/branch}",
      "collaborators_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/collaborators{/collaborator}",
      "comments_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/comments{/number}",
      "commits_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/commits{/sha}",
      "compare_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/compare/{base}...{head}",
      "contents_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/contents/{+path}",
      "contributors_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/contributors",
      "deployments_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/deployments",
      "description": "an effort to correct various shortcomings of the default GitHub action and event system",
      "downloads_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/downloads",
      "events_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/events",
      "fork": false,
      "forks_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/forks",
      "full_name": "gofunky/interactive-event-dispatcher",
      "git_commits_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/git/commits{/sha}",
      "git_refs_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/git/refs{/sha}",
      "git_tags_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/git/tags{/sha}",
      "hooks_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/hooks",
      "html_url": "https://github.com/gofunky/interactive-event-dispatcher",
      "id": 300029405,
      "issue_comment_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/issues/comments{/number}",
      "issue_events_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/issues/events{/number}",
      "issues_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/issues{/number}",
      "keys_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/keys{/key_id}",
      "labels_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/labels{/name}",
      "languages_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/languages",
      "merges_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/merges",
      "milestones_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/milestones{/number}",
      "name": "interactive-event-dispatcher",
      "node_id": "MDEwOlJlcG9zaXRvcnkzMDAwMjk0MDU=",
      "notifications_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/notifications{?since,all,participating}",
      "owner": {
        "avatar_url": "https://avatars3.githubusercontent.com/u/41548062?v=4",
        "events_url": "https://api.github.com/users/gofunky/events{/privacy}",
        "followers_url": "https://api.github.com/users/gofunky/followers",
        "following_url": "https://api.github.com/users/gofunky/following{/other_user}",
        "gists_url": "https://api.github.com/users/gofunky/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/gofunky",
        "id": 41548062,
        "login": "gofunky",
        "node_id": "MDEyOk9yZ2FuaXphdGlvbjQxNTQ4MDYy",
        "organizations_url": "https://api.github.com/users/gofunky/orgs",
        "received_events_url": "https://api.github.com/users/gofunky/received_events",
        "repos_url": "https://api.github.com/users/gofunky/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/gofunky/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/gofunky/subscriptions",
        "type": "Organization",
        "url": "https://api.github.com/users/gofunky"
      },
      "private": false,
      "pulls_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/pulls{/number}",
      "releases_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/releases{/id}",
      "stargazers_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/stargazers",
      "statuses_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/statuses/{sha}",
      "subscribers_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/subscribers",
      "subscription_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/subscription",
      "tags_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/tags",
      "teams_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/teams",
      "trees_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/git/trees{/sha}",
      "url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher"
    },
    "rerun_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/actions/runs/289782451/rerun",
    "run_number": 163,
    "status": "completed",
    "updated_at": "2020-10-05T16:33:49Z",
    "url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/actions/runs/289782451",
    "workflow_id": 2823525,
    "workflow_url": "https://api.github.com/repos/gofunky/interactive-event-dispatcher/actions/workflows/2823525"
  }
```

</details>

**Comment by gr2m** [comment]
@matfax the event types are generated based on https://github.com/octokit/webhooks. It seems that we don't have a `workflow_run` payload yet, could send a PR to add yours at https://github.com/octokit/webhooks/tree/master/payload-examples/api.github.com? That should resolve the problem. Once you add the payload, run 

```
node bin/octokit-webhooks.js update --cached
```

to update the exported [`index.json`](https://github.com/octokit/webhooks/blob/master/index.json)

**Comment by oscard0m** [comment]
Doing a follow up to validate if this issue was already fixed. @matfax

**Comment by matfax** [comment]
@dominguezcelada `@octokit\webhooks\dist-types\generated\event-payloads.d.ts` of latest NPM version `7.12.2` doesn't have the new fields yet.

**Comment by oscard0m** [comment]
> @dominguezcelada `@octokit\webhooks\dist-types\generated\event-payloads.d.ts` of latest NPM version `7.12.2` doesn't have the new fields yet.

Yep, seems that the commit with the bump of @octokit/webhooks to 3.18.0 has not been released yet. In next release it will be published.

@gr2m is it possible to trigger a release here so publish the patch of this issue?

Thanks 🙏🏽

**Comment by oscard0m** [comment]
@matfax 7.13.0 was released with the upgrade. Could you double check when you have time if the issue was solved?

**Comment by matfax** [comment]
I confirm that it is resolved with version `7.13.0`. Thanks for your support.
